controller/dataplane/owned_deployment.go

```diff
@@ -306,17 +318,55 @@ func reconcileDataPlaneDeployment(
 
 		k8sresources.SetDefaultsPodTemplateSpec(&desired.Spec.Template)
 
+		// Keep track of this for logging purposes
+		originalReplicaCount := existing.Spec.Replicas
+
 		// ensure that object metadata is up to date
 		updated, existing.ObjectMeta = k8sutils.EnsureObjectMetaIsUpdated(existing.ObjectMeta, desired.ObjectMeta)
 
-		// some custom comparison rules are needed for some PodTemplateSpec sub-attributes, in particular
-		// resources and affinity.
+		// Check if this is a deployment restart (kubectl rollout restart)
+		// We need to check if the existing deployment has the restart annotation
+		// This is different from our previous approach because the desired deployment
+		// is generated from the DataPlane resource and won't have the restart annotation
+		isRestartOperation := false
+
+		// Check if we have a fresh restart annotation (from the last 5 minutes)
+		// This prevents us from treating all deployments with old restart annotations
+		// as perpetually in a restart state, fixing issue #1390
+		hasRestartAnnotation, restartTimeStr := podTemplateSpecHasRestartAnnotation(&existing.Spec.Template)
+		if hasRestartAnnotation {
+			// Only treat it as a restart if the timestamp is recent (within 5 minutes)
+			restartTime, err := time.Parse(time.RFC3339, restartTimeStr)
+			if err == nil {
+				// Check if restart annotation is less than 5 minutes old
+				if time.Since(restartTime) < 5*time.Minute {
```

Comment:
How did we establish the 5 min as a threshold here? Was it suggested by the community (other operators do the same?) or it's a reasonable estimate that we believe would work well here?

Reply:
Got it. So I believe we could leave a comment here about the chosen value: something like "We use 5m here as threshold as that correlates with const used by k/k (<PROVIDE LINK HERE>). Restart annotations older than this duration shouldn't cause a reconciliation effect". Or similar.